Tensile/Tests/unit/test_mergeLogic.py

```diff
@@ -0,0 +1,180 @@
+uniqueSolution=logicPrefix+r"""
+-
+  - SolutionIndex: 0
+    SolutionNameMin: Kernel0
+  - SolutionIndex: 1
+    SolutionNameMin: Kernel1
+"""
+
+notTrimmedSize=r"""
+-
+  - - [128, 128, 1, 128, 128, 128, 128, 128]
+    - [1, 999.9]
+  - - [1024, 1024, 1, 1024]
+    - [42, 0.0]
+  - - [128, 128, 1, 128]
+    - [39, 0.0]
+  - - [1024, 1024, 1, 1024, 1024, 1024, 1024, 1024]
+    - [39, 999.9]
+  - - [1024, 1024, 1, 1024, 1044, 1044, 1044, 1044]
+    - [39, 999.9]
+"""
+
+trimmedSize=r"""
+-
+  - - [128, 128, 1, 128]
+    - [1, 999.9]
+  - - [512, 512, 1, 512]
+    - [39, 999.9]
+  - - [1024, 1024, 1, 1024]
+    - [42, 999.9]
+"""
+
+def checkUniqueSolution(solutionPool):
+  uniq = set()
+  # note: any([False or None, True or None]) -> True
+  return not any(frozenset(cmpHelper(i).items()) in uniq or uniq.add(frozenset(cmpHelper(i).items()))
+                 for i in solutionPool)
+
+@pytest.mark.parametrize("sizes, expected", [
+  (   trimmedSize, [[128,128,1,128],[512,512,1,512],[1024,1024,1,1024]]),
+  (notTrimmedSize, [[1024,1024,1,1024],[128,128,1,128]])
+  ])
+def test_fixSizeInconsistencies(sizes, expected):
+  data = yaml.load(sizes, yaml.SafeLoader)
+  data_ = fixSizeInconsistencies(data[0], "dummy")
+
+  for [size, [_,_]], expected_ in zip(data_[0], expected):
+    assert size == expected_
+  # print final yaml for visual inspection
+  # stream = io.StringIO("")
+  # yaml.safe_dump(data_, stream, default_flow_style=None)
+  # print(stream.getvalue())
+
+@pytest.mark.parametrize("input,expectedNumKernelRemoved", [(baseLogic, 1),
+                                                            (incLogic, 0)])
+def test_removeUnusedKernels(input, expectedNumKernelRemoved):
+  data = yaml.load(input, yaml.SafeLoader)
+  dataFiltered, numKernelRemoved = removeUnusedKernels(data)
+
+  # test if number of solution removed is correct
+  assert numKernelRemoved == expectedNumKernelRemoved
+
+  # test if solution is re-indexed
+  for index, s in enumerate(dataFiltered[5]):
+    assert index == s["SolutionIndex"]
+
+  # check if size-kernel mapping is not compromised
+  solutionMapBefore = {}
+  solutionMapAfter = {}
+  for [size, [index, _]] in data[7]:
+    solutionMapBefore[tuple(size)] = [s["SolutionNameMin"] for s in data[5] if s["SolutionIndex"]==index][0]
+  for [size, [index, _]] in dataFiltered[7]:
+    solutionMapAfter[tuple(size)] = [s["SolutionNameMin"] for s in dataFiltered[5] if s["SolutionIndex"]==index][0]
+
+  assert solutionMapBefore == solutionMapAfter
+
+  # print final yaml for visual inspection
+  # stream = io.StringIO("")
+  # yaml.safe_dump(dataFiltered, stream, default_flow_style=None)
+  # print(stream.getvalue())
+
+def test_mergeLogic():
+  baseData = yaml.load(baseLogic, yaml.SafeLoader)
+  incData = yaml.load(incLogic, yaml.SafeLoader)
+
+  mergedData, *stats = mergeLogic(baseData, incData, False)
+
+  # check if stats are as expected
+  assert stats == [1,2,2] # 1 sizes added, 2 kernels added, 2 kernels removed/replaced
```

Comment:
what are the 2 kernels removed/replaced?
1 removed: 
  * baseLogic - "UnusedSolution"

2 replaced: 
  * [128, 128, 1, 128] from "InUseForSize128or64" ->"InUseForSize128xxx" and 
  * [256, 256, 1, 256] from "InUseForSize256" ->"InUseForSize256or1024xxx"

Reply:
In this case, the kernel "InUseForSize128or64" coexists with kernel "InUseForSize128xxx" because the former kernel is still referenced by gemm size [64, 64, 1, 64] so it's not counted as being replaced. That makes:

- 1 removed: "UnusedSolution"
- 1 replaced: "InUseForSize256"

So the total count is 2

I appreciate your detailed review